Public/Src/Cache/ContentStore/Library/Service/LocalContentServerBase.cs

```diff
@@ -78,11 +85,8 @@ public abstract class LocalContentServerBase<TStore, TSession> : StartupShutdown
         /// </summary>
         /// <remarks>
         /// This is only supposed to be used by this class and inheritors, do not make internal or public.
-        /// 
-        /// It is also expected to be immutable, because inheritors may copy it around. So do NOT add keys after
-        /// constructed.
         /// </remarks>
-        protected readonly Dictionary<string, TStore> StoresByName = new Dictionary<string, TStore>();
+        public readonly IReadOnlyDictionary<string, TStore> StoresByName;

```

Comment:
What I think you're trying to do here is a good idea, but this is an incomplete implementation of that idea. Any arbitrary code can still read this field, cast to Dictionary, and modify entries. The complete way to do this is to create a private dictionary field, a public IReadOnlyDictionary property, and have the property return a ReadOnlyDictionary that wraps the private dictionary. Extra points for making sure the wrapper is only constructed once, not on every call.